packages/@react-stately/grid/src/useGridState.ts

```diff
@@ -17,15 +17,18 @@ export interface GridState<T, C extends GridCollection<T>> {
 export interface GridStateOptions<T, C extends GridCollection<T>> extends MultipleSelectionStateProps {
   collection: C,
   disabledKeys?: Iterable<Key>,
-  focusMode?: 'row' | 'cell'
+  focusMode?: 'row' | 'cell',
+  /** @private - do not use unless you know what you're doing. */
+  UNSAFE_selectionState?: MultipleSelectionState
```

Comment:
Don't love this, but we need to avoid a circular dependency in RAC `Table`. We need the selection state in order to determine whether to render checkboxes based on the selectionBehavior etc (via `useTableOptions`), but we can't build the full `TableState` until we have the collection. This prop lets us split this up to call `useMultipleSelectionState` directly, and pass this state down into `useTableState`. It is unsafe because of the rules of hooks below, so this is not meant to be a public API. Happy to change if someone has a better idea.

Reply:
Would it make more sense for useMultipleSelectionState to have this and reconcile the difference? then if we run into this issue in other places it's not just in useGridState?

Also, we could not call it conditionally below, it just sets up state and then if nothing interacts with it, it won't do anything. So should be harmless to call it unconditionally and only use the result if props.UNSAFE_selectionState isn't passed in.

Reply:
yeah maybe. It bothers me to do extra unnecessary work and throw it away lol